File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;


int gcd(int a, int b)
{
    int c;
	while(b)
	{
		c = a % b;
		a = b;
		b = c;
	}
    return a;
}

int lcm(int num1, int num2) {
    int s, b;
    if (num1 < num2) {
        s = num1;
        b = num2;
    } else {
        s = num2;
        b = num1;
    }
    return s*b/gcd(s,b);
}

int solution(vector<int> arr) {

    int ans = 1;
    for (auto &num : arr) {
        ans = lcm(num, ans);
    }
    return ans;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
from math import gcd # 최대공약수를 찾는 함수

def solution(arr):
    lcm = arr[0] # 첫번째 원소
    for x in arr[1:]: # 두번째 원소부터 기존 lcm과 연산
        lcm = (lcm * x) // gcd(lcm, x)

    return lcm